autogpt_platform/backend/backend/data/model.py

```diff
@@ -164,16 +166,30 @@ def CredentialsField(
     `CredentialsField` must and can only be used on fields named `credentials`.
     This is enforced by the `BlockSchema` base class.
     """
+    # Convert provider list to enum for schema
+    provider_list = [provider] if isinstance(provider, str) else provider
+
     json_extra = {
         k: v
         for k, v in {
-            "credentials_provider": provider,
+            "credentials_provider": (
+                provider if isinstance(provider, str) else list(provider)
+            ),
             "credentials_scopes": list(required_scopes) or None,  # omit if empty
             "credentials_types": list(supported_credential_types),
+            "discriminator": discriminator,
+            "discriminator_mapping": discriminator_mapping,
         }.items()
         if v is not None
     }
 
+    # Force provider to always be a string
+    # Otherwise input provider will fail jsonschema validation
+    # if multiple providers are provided
+    json_extra.update(
+        {"properties": {"provider": {"type": "string", "enum": provider_list}}}
+    )
+
```

Comment:
I think you can avoid this hack by changing the `provider` parameter to `provider: CP | list[CP]` and change the definition of `CP` back to `TypeVar("CP", bound=str)`